.github/pull_request_template.md

```diff
@@ -1,25 +1,36 @@
-# Description
+## Description
 
-Please include a summary of the change.
+*Replace this paragraph with a description of what this PR is doing. If you're modifying existing behavior, describe the existing behavior, how this MR is changing it, and what motivated the change.*
 
-Fixes # (issue number, if there is one)
+## Checklist
 
-## Type of change
+Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.
 
-Please delete options that are not relevant.
+- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
+- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
+- [ ] All existing and new tests are passing.
+- [ ] I updated/added relevant documentation (doc comments with `///`).
+- [ ] The Flame analyzer (`./scripts/analyze.sh`) does not report any problems on my PR.
+- [ ] I have added examples for new features in `doc/examples`.
+- [ ] I have formatted my code with `./scripts/format.sh`.
+- [ ] I read and followed the [Flutter Style Guide].
+- [ ] I have added a description of the change under `[next]` in `CHANGELOG.md`.
+- [ ] I am willing to follow-up on review comments in a timely manner.
+- [ ] I am done with this PR and removed the `Draft` status, by clicking on the `Read for review` button in this PR.
```

Comment:
I am done sounds a bit too profound here, how about something like:
`I am happy with the current version of this PR to be reviewed` or something